### ISSUE TITLE
NPE on handoverQueryPointer

### ISSUE THREAD

**Issue by ppamorim** [realm/realm-java#4086]
New random bug happended, this problem can be related with the #2021.

```java
java.lang.NullPointerException: Attempt to invoke virtual method 'long io.realm.RealmQuery.handoverQueryPointer()' on a null object reference
  at io.realm.HandlerController.completedAsyncRealmObject(HandlerController.java:653)
  at io.realm.HandlerController.handleMessage(HandlerController.java:132)
  at android.os.Handler.dispatchMessage(Handler.java:98)
  at android.os.Looper.loop(Looper.java:135)
  at android.app.ActivityThread.main(ActivityThread.java:5343)
  at java.lang.reflect.Method.invoke(Native Method)
  at java.lang.reflect.Method.invoke(Method.java:372)
  at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:905)
  at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:700)
```

- Realm 2.3.0
- Kotlin 1.0.4
- Gradle 2.14.1

**Comment by Zhuinden** [comment]
Sounds related yes, I still think the temporary workaround would be to use the synchronous query api and wait for the results integration for async queries to work reliably again

**Comment by beeender** [comment]
Should be fixed by #3834

**Comment by beeender** [comment]
#3834 is merged to master. will be fixed in `v2.4.0`.
